crates/oxc_linter/src/rules/no_empty.rs

```diff
@@ -0,0 +1,110 @@
+use oxc_ast::{AstKind, Span};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-empty): Disallow empty block statements")]
+#[diagnostic(severity(warning), help("Add comment inside empty {0} statement"))]
+struct NoEmptyDiagnostic(&'static str, #[label("Empty {0} statement")] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoEmpty {
+    allow_empty_catch: bool,
+}
+
+const RULE_NAME: &str = "no-empty";
+
+impl Rule for NoEmpty {
+    const NAME: &'static str = RULE_NAME;
+
+    fn from_json(value: serde_json::Value) -> Self {
```

Comment:
its not immediately clear to me what this does exactly?

Reply:
Good catch, let me think about this for a bit. It's a constructor for initializing from a json configuration.